### ISSUE TITLE
Couldn't find any compatible video codecs [Bug on Linux (Usb Version)] 

### ISSUE THREAD

**Issue by shitwooboo** [exelix11/SysDVR#151]
When running dotnet SysDVR-Client.dll usb,I get the following error on a linux laptop. Please help me resolve this issue. 

Using dotnet SysDVR-Client.dll usb --rtsp while the switch is connected over usb works fine. It opens a network stream which I can play over vlc. However, I am trying to get the usb stream to work so that there is less lag and stutter in the stream. 

![Screenshot from 2021-08-27 03-28-04](https://user-images.githubusercontent.com/89631359/131090336-d16260af-c16c-457e-bf52-556aa121eef7.png)

**Comment by shitwooboo** [comment]
This is what I get from doing that
![Screenshot from 2021-08-27 19-07-28](https://user-images.githubusercontent.com/89631359/131196873-c7595e7f-8dd6-456d-952c-6b600da45d88.png)

**Comment by shitwooboo** [comment]
I also don't see a avcodec.58 file in my x86_x64-linux gnu folder. I only see an avcodec.so.58 if that helps. I also created a synlink to avcodec.so.58 from libavcodec.so.57 to solve an earlier issue if that helps.

**Comment by bleck9999** [comment]
If you linked avcodec.so.58 to libavcodec.so.57 it's possible that av_codec_iterate doesn't exist in libavcodec 57 but does in 58, so try actually installing libavcodec 58.

**Comment by shitwooboo** [comment]
Thank you, do you have any idea how to install libavcodec 58 on linux, I've tried looking online but I don't know how to do it. I've also tried removing libavcodec-dev and reinstalling it, but I only see libavcodec 57

**Comment by shitwooboo** [comment]
If anyone has a similar issue, even though doing ffmpeg -version showed that I had libavcodec 58 installed for both avcodec and avformat, I don't think it was entirely correct and might have been misleading. I removed my earlier synlinks and then I added this repo and did an upgrade as shown in the link below.
https://ubuntuhandbook.org/index.php/2021/05/install-ffmpeg-4-4-ppa-ubuntu-20-04-21-04/
After doing this it started working.

Earlier I also added
/usr/local/lib
/usr/lib
into the bottom of ld.so.conf (located in etc/) (but I'm not sure if this was necessary)
and then did sudo lconfig to update the changes.
